Plugin.pm

```diff
@@ -1208,8 +1257,29 @@ sub _trackItem {
 		$item->{line1} .= ' (' . cstring($client, 'PLUGIN_QOBUZ_HIRES') . ')';
 	}
 
-	if ( $track->{work} ) {
-		$item->{work} = $track->{work};
+	# Enhancements to work/composer display for classical music (tags returned from Qobuz are all over the place)
+	my $genreList = $prefs->get('classicalGenres');
+	if ( $prefs->get('useClassicalEnhancements')
+		&& ( grep(/Classique/,@{$track->{album}->{genres_list}}) || grep(/(^\s*$genre\s*$|^\s*\*$)/,(split ',', $genreList )) ) 
+	   ) 
+	{
```

Comment:
Hmm... tricky formatting here 😁. 

I'm not sure I understand the second part of the regex. Is this to match empty entries? If that was the case then you should probably make sure you remove empty values before storing the string in the prefs. And you should probably split by a regex `/\s*,\s*` instead of just the comma to get rid of surrounding spaces.

Reply:
And I'm not sure whether you've tested for the existence of `$track->{album}->{genre_list}` before. It might be worth a check before trying to access it.

Reply:
The second part of the regex was a hangover from before I identified that the Qobuz tag genre_list=Classique could drive the whole thing - the previous version allowed the user to add a genre of "*" in the preferences if they were overwhelmingly a classical listener and didn't want to spend the rest of their lives adding Qobuz genre tags to their list. Removed.

Using your suggested regex in the split instead of in the grep doesn't work when an additional genre entered by the user contains spaces. For example, if the user enters an additional genre of "Vocal Jazz", it will then match the Qobuz genre of "Jazz". Unlikely, but you know...

I've added a check for the existence of $track->{album}->{genre_list} to the code:
`&& ( $track->{album}->{genres_list} && grep(/Classique/,@{$track->{album}->{genres_list}}) || grep(/^\s*$genre\s*$/,(split ',', $genreList )) ) `

Reply:
my suggestion would still require a space to be there as the separator. But it would accept " , " as well as "," or ", ".

Reply:
~Two~ A few more comments on that function:

- I'd return early if `useClassicalEnhancements` was not set. Makes the `if` statement more readable: `return if !$prefs->get('useClassicalEnhancements');` as the second statement in the function. Also saves splitting genres in this case.
- `classicalGenres` could be undefined, couldn't it? In that case the `split` would complain. Make sure that pref exists before trying to split it.
- If you use a variable in a regex you should "quote" it: `/^\s*\Q$genre\E\s*$/`. Otherwise a value like `Mu$ic` would crash LMS or otherwise behave badly (https://perldoc.perl.org/perlre#Quoting-metacharacters)
- if that method was called often you could map the `@genreList` to a hash with the entries as the key and a truthy value. It then would be much more efficient looking up a genre by doing `$genreList{$genre}` instead of grepping using a regex over and over again.

Reply:
classicalGenres is defined, but might be empty, which I know is a boolean false in Perl - is that the same as undefined? In any case, I've just tested with it an empty preference, and there are no complaints in the log.

Reply:
Empty is not the same as undefined, no. If you initialize the pref with an empty value, you should be good.